Move config docs to contributing guide, kill `docs` directory.

jchristgit opened the pull request:
Continuation of #521.

jchristgit committed:
Move configuration section to site contributing guide.

jchristgit committed:
Kill `docs/` directory.

kosayoda commented on pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md:
```diff
@@ -143,3 +143,35 @@ Unless you are editing the Dockerfile or docker-compose.yml, you shouldn't need
+  secret as the name suggests! This is managed for you in debug setups.
+
+- **`STATIC_ROOT`**: The root in which `python manage.py collectstatic` collects
+  static files. Optional, defaults to `/var/www/pythondiscord.com`.
```
Is this still true given our current `STATIC_ROOT` default?

https://github.com/python-discord/site/blob/efa3c450f7cabaac5191c73c6f4726a1be89be23/pydis_site/settings.py#L168

jchristgit replied:
Hello, good point, no, that's outdated, you're correct. This is from the very early pre-Docker times. I will update it.

jchristgit replied:
See ef59812d.

kosayoda reviewed (commented):
Can we have a link mentioning this new details section from the `2. Environment variables` section?

jchristgit committed:
Address review comments.

- Link detail section in environment variables section.
- Update old static path from pre-Docker environment times and mention the
  Docker setup.
diff --git
a/pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md
b/pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md
index 8c21b5b0..24227f24 100644 ---
a/pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md
+++
b/pydis_site/apps/content/resources/guides/pydis-guides/contributing/site.md @@
-79,6 +79,9 @@ SECRET_KEY=suitable-for-development-only STATIC_ROOT=staticfiles
```

+The [Configuration in Detail](#configuration-in-detail) section contains
+detailed information about these settings.
+
 #### Notes regarding `DATABASE_URL`

 - If the database is hosted locally i.e. on the same machine as the webserver, then use `localhost` for the host. Windows and macOS users may need to use the [Docker host IP](../hosts-file/#windows) instead.
@@ -173,5 +176,6 @@ The website is configured through the following environment variables:
 - **`SECRET_KEY`**: The secret key used in various parts of Django. Keep this
   secret as the name suggests! This is managed for you in debug setups.

-- **`STATIC_ROOT`**: The root in which `python manage.py collectstatic` collects
-  static files. Optional, defaults to `/var/www/pythondiscord.com`.
+- **`STATIC_ROOT`**: The root in which `python manage.py collectstatic`
+  collects static files. Optional, defaults to `/app/staticfiles` for the
+  standard Docker deployment.

kosayoda committed:
Merge branch 'main' into move-config-docs-to-contributing-guide
